lib/valueflow.cpp

```diff
@@ -488,7 +488,7 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
 
     // cast..
     if (const Token *castType = getCastTypeStartToken(parent)) {
-        if (astIsPointer(tok) && value.valueType == ValueFlow::Value::INT &&
+        if ((tok->valueType() == nullptr || (astIsPointer(tok) && value.valueType == ValueFlow::Value::INT)) &&
```

Comment:
Maybe this should be `tok->valueType() == nullptr  && value.isImpossible()`

Reply:
Actually when `tok->valueType()` is null we should still check that the `value.valueType` is an INT. This should probably be `((tok->valueType() == nullptr && value.isImpossible()) || astIsPointer(tok)) && value.valueType == ValueFlow::Value::INT`

Reply:
I believe I've made the change you're suggesting. I admit I don't follow it but it passes the tests.